### ISSUE TITLE
Ask for training code

### ISSUE THREAD

**Issue by hjwdzh** [torrvision/crfasrnn#28]
Hi Zhengshuai,

I am a first year phd student in stanford called Jingwei Huang. I am very interested in your work and want to use your architecture for further research. Can you tell me how I can train the data?

**Comment by bittnt** [comment]
@hjwdzh 
I believe you should check how to train FCN network from this.

To train our network, just append the multistagemeanfield layer at the bottom of the script:
'''
layer { type: "Deconvolution" name: 'upsample-8' bottom: 'score-fr' top: 'bigscore' param { lr_mult: 0 } convolution_param { bias_term: false num_output: 21 kernel_size: 16 stride: 8 } } 

layer { type: "Split" name: 'splitting' bottom: 'bigscore' top: 'unary' top: 'Q0' } 

layer { name: "inference1" type: "MultiStageMeanfield" bottom: "unary" bottom: "Q0" bottom: "datamf" top: "upscore" param { lr_mult: 10000 } param { lr_mult: 10000 } param { lr_mult: 1000 } multi_stage_meanfield_param { num_iterations: 5 compatibility_mode: POTTS threshold: 2 theta_alpha: 160 theta_beta: 3 theta_gamma: 3 spatial_filter_weight: 3 bilateral_filter_weight: 5 } } 

layer { name: "loss" type: "SoftmaxWithLoss" bottom: "upscore" bottom: "label" top: "loss" loss_param { ignore_label: 255 normalize: false } include: { phase: TRAIN } } 
'''

**Comment by martinkersner** [comment]
I have a question related to learning rate inside of *MultiStageMeanfield*. In *TVG_CRFRNN_COCO_VOC.prototxt* you specify learning rates with values 0.001, 0.001 and 0.01. However, here you posted much higher learning rates 10000, 10000 and 1000. Using higher values I am able to achieve better results, however, I am still wondering which one did you use for training with base learning rate 1e-13 which is mentioned in paper.

**Comment by bittnt** [comment]
Glad to hear that you get better performance.

Setting up learning rate higher for this last layer , e.g. 1000 or 10000
helps to get top performance, within our end to end framework and base lr
at 1e-13.

For other purpose like that you might want to fine tune the network a bit
more within crf-rnn, in which case you can set them to 0.001 or some small
number.

We will work out public available simple scripts to do this soon.

**Comment by sadeepj** [comment]
Closing old issues with no recent activity.
